src/forcing_component.cpp

```diff
@@ -340,7 +340,10 @@ void ForcingComponent::run( const double runToDate ) throw ( h_exception ) {
         H_LOG( logger, Logger::DEBUG ) << "forcing total is " << forcings[ D_RF_TOTAL ] << std::endl;
         
         //---------- Change to relative forcing ----------
-        // At this point, we've computed all absolute forcings. If base year, save those values
+        // Note that the code below assumes model is always consistently run from base-year forward. 
+        // Results will not be consistent if parameters are changed but base-year is not re-run.
```

Comment:
This is true throughout the model; if you change a parameter, you have to call `reset`.  In the R interface we guard this and force the reset if necessary before a run.  In the C++ interface it's up to the caller to make sure this happens.

I think the more important observation here is that by default `baseyear` is `getStartDate()`, but we allow it to be set manually. In fact, we do this in the default input files; the start year is 1745, but we set the base year for forcing to 1750.  Forcing is automatically reported as zero in years prior to the base year, so that's ok, but there is nothing in the code to prevent setting the the forcing base year to something _prior_ to the start year, which would probably fail spectacularly.

Reply:
It's ready. This is the first step for other work, but this part is complete re: the new reporting functionality.

Reply:
Ok, I'll take it for a spin this afternoon.